qt/scientific_interfaces/Inelastic/BayesFitting/StretchView.cpp

```diff
@@ -0,0 +1,326 @@
+  m_uiForm.dsResolution->setWSSuffixes(filter ? getResolutionWSSuffixes(tabName) : noSuffixes);
+}
+
+void StretchView::setLoadHistory(bool doLoadHistory) {
+  m_uiForm.dsSample->setLoadProperty("LoadHistory", doLoadHistory);
+  m_uiForm.dsResolution->setLoadProperty("LoadHistory", doLoadHistory);
+}
+
+void StretchView::validateUserInput(IUserInputValidator *validator) const {
+  validator->checkDataSelectorIsValid("Sample", m_uiForm.dsSample);
+  validator->checkDataSelectorIsValid("Resolution", m_uiForm.dsResolution);
+}
+
+StretchRunData StretchView::getRunData() const {
+  auto const useQuickBayes = SettingsHelper::hasDevelopmentFlag("quickbayes");
+
+  auto const sampleName = m_uiForm.dsSample->getCurrentDataName().toStdString();
+  auto const resName = m_uiForm.dsResolution->getCurrentDataName().toStdString();
+
+  auto const background = m_uiForm.cbBackground->currentText().toStdString();
+
+  auto const eMin = m_properties["EMin"]->valueText().toDouble();
+  auto const eMax = m_properties["EMax"]->valueText().toDouble();
+  auto const beta = m_properties["Beta"]->valueText().toInt();
+
+  auto const elasticPeak = m_uiForm.chkElasticPeak->isChecked();
+
+  auto const sigma = !useQuickBayes ? m_properties["Sigma"]->valueText().toInt() : 0;
+  auto const nBins = !useQuickBayes ? m_properties["SampleBinning"]->valueText().toInt() : 0;
+  auto const sequence = !useQuickBayes ? m_uiForm.chkSequentialFit->isChecked() : false;
+
+  return StretchRunData(sampleName, resName, eMin, eMax, beta, elasticPeak, background, sigma, nBins, sequence);
+}
+
+CurrentPreviewData StretchView::getCurrentPreviewData() const {
+  auto const sampleName = m_uiForm.dsSample->getCurrentDataName().toStdString();
+  auto const hasSample = m_uiForm.ppPlot->hasCurve("Sample");
+
+  return CurrentPreviewData(sampleName, hasSample);
+}
+
+std::string StretchView::getPlotType() const { return m_uiForm.cbPlot->currentText().toStdString(); }
+
+std::string StretchView::getPlotContour() const { return m_uiForm.cbPlotContour->currentText().toStdString(); }
+
+IRunView *StretchView::getRunWidget() const { return m_uiForm.runWidget; }
+
+void StretchView::resetPlotContourOptions(const std::vector<std::string> &contourNames) {
+  m_uiForm.cbPlotContour->clear();
+  for (auto const &name : contourNames)
+    m_uiForm.cbPlotContour->addItem(QString::fromStdString(name));
+}
+
+void StretchView::setPlotADSEnabled(bool enabled) { m_uiForm.ppPlot->watchADS(enabled); }
+
+void StretchView::setPlotResultEnabled(bool enabled) {
+  m_uiForm.pbPlot->setEnabled(enabled);
+  m_uiForm.cbPlot->setEnabled(enabled);
+}
+
+void StretchView::setPlotContourEnabled(bool enabled) {
+  m_uiForm.pbPlotContour->setEnabled(enabled);
+  m_uiForm.cbPlotContour->setEnabled(enabled);
+}
+
+void StretchView::setSaveResultEnabled(bool enabled) { m_uiForm.pbSave->setEnabled(enabled); }
+
+int StretchView::displaySaveDirectoryMessage() {
```

Comment:
Can this return type be `bool`? I think at the moment its doing an implicit cast from bool to int